### PR TITLE
JW-1122 controller cleanup2

### PR DESCRIPTION
**Pull request by mclapinski**

### CODE REVIEW

File: src/base/misc_utils.cc [hunk 1]
```diff
@@ -136,20 +137,22 @@ bool MiscUtils::GetContrailVersionInfo(BuildModule id, string &rpm_version,
...
-    if (hModule) {
-        if (LoadString(hModule, IDS_CONTRAIL_RPM_VERSION, buffer, bufsize) > 0) {
-            rpm_version = buffer;
-            if (LoadString(hModule, IDS_CONTRAIL_BUILD_NUM, buffer, bufsize) > 0) {
-                build_num = buffer;
-                ret = true;
-            }
-        }
-    }
+    if (!hModule)
```
Comment (sodar) [inline src/base/misc_utils.cc]: We could compare this to `NULL`, instead of checking for zero.

File: src/base/misc_utils.cc [hunk 2]
```diff
@@ -184,11 +185,43 @@ bool MiscUtils::GetBuildInfo(BuildModule id, const string &build_info,
...
+    std::stringstream ss;
+
+    // structure used to store enumerated languages and code pages
+    struct LANGANDCODEPAGE {
+        WORD wLanguage;
+        WORD wCodePage;
+    } *lpTranslate;
+
+    size = GetFileVersionInfoSize(filename, NULL);
+    data = new BYTE[size];
```
Comment (sodar) [inline src/base/misc_utils.cc]: `data` buffer is not freed in this function.

File: src/vnsw/agent/port_ipc/rest_common.cc [hunk 3]
```diff
@@ -26,12 +26,8 @@ void SendResponse(HttpSession *session,
 void SendErrorResponse(HttpSession *session,
                         const std::string &error_msg, int status_code) {
     rapidjson::Document document;
-	//rapidjson::Value v1,v2;
     document.SetObject();
-	//WINDOWS-TEMP
-	rapidjson::GenericStringRef<char> sr1("error"), sr2(error_msg.c_str());
-    //document.AddMember("error", error_msg.c_str(), document.GetAllocator());
-	document.AddMember(sr1, sr2, document.GetAllocator());
+	document.AddMember("error", error_msg.c_str(), document.GetAllocator());
```
Comment (sodar) [inline src/vnsw/agent/port_ipc/rest_common.cc]: Could you remove this tab?

**Review by sodar — CHANGES_REQUESTED** [review]

File: src/base/misc_utils.cc [hunk 4]
```diff
@@ -118,13 +118,14 @@ bool MiscUtils::GetVersionInfoInternal(const string &cmd, string &rpm_version,
 }
 #endif
 
-bool MiscUtils::GetContrailVersionInfo(BuildModule id, string &rpm_version,
-    string &build_num) {
+bool MiscUtils::GetContrailVersionInfo(BuildModule id, string &rpm_version,
```
Comment (jacek-izykowski) [inline src/base/misc_utils.cc:121]: Useless space.

Comment (mclapinski) [reply src/base/misc_utils.cc:121]: It's their space. I just restored it.

File: src/base/misc_utils.cc [hunk 5]
```diff
@@ -184,11 +185,45 @@ bool MiscUtils::GetBuildInfo(BuildModule id, const string &build_info,
...
+    LPVOID lpBuffer;
+    std::stringstream ss;
+
+    // structure used to store enumerated languages and code pages
+    struct LANGANDCODEPAGE {
+        WORD wLanguage;
+        WORD wCodePage;
+    } *lpTranslate;
+
+    size = GetFileVersionInfoSize(filename, NULL);
```
Comment (jacek-izykowski) [inline src/base/misc_utils.cc:206]: You can return false, when GetFileVersionInfoSize fails.

File: src/base/misc_utils.cc [hunk 6]
```diff
@@ -184,11 +185,45 @@ bool MiscUtils::GetBuildInfo(BuildModule id, const string &build_info,
...
+       << lpTranslate[0].wLanguage << std::setw(4) << lpTranslate[0].wCodePage
+       << "\\ProductVersion";
+
+    if (!VerQueryValue(data, ss.str().c_str(), &lpBuffer, &dwBytes))
+        return false;
+
+    distro = "Windows";
+    code_name = (LPTSTR)lpBuffer;
+
+    delete[] data;
```
Comment (jacek-izykowski) [inline src/base/misc_utils.cc]: There are still a few execution paths, where this is not called.

**Review by jacek-izykowski — CHANGES_REQUESTED** [review]
MiscUtils::GetPlatformInfo requires some refinements (see remarks).

**Review by sodar — CHANGES_REQUESTED** [review]
Besides @jacek-izykowski comment, the rest looks good to me

**Review by sodar — APPROVED** [review]

**Review by jacek-izykowski — APPROVED** [review]

**Pull request merged**